tools/istio-iptables/pkg/builder/iptables-builder.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package builder
+
+// Rule represents iptables rule - chain, table and options
+type Rule struct {
```

Comment:
Please move the `Rule` and `Rules` types out of this file. They are implementation details that are not (and should not be) referred to by the interface definitions.

Reply:
done